modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/itemActions.spec.ts

```diff
@@ -47,111 +56,663 @@ test.afterEach(async ({dataSetManagerApiHelpers}) => {
+				page,
+				text: "This key is used to display the action if the user has the permission for it by checking if the key is present in the data item's actions array.",
+			});
+
+			await checkHelperTooltip({
+				formElement: form.confirmationMessageInput,
+				page,
+				text: 'The user will see this message before performing the action. No message will be displayed if the message is empty.',
+			});
+
+			await form.typeSelect.selectOption('Async');
+
+			await actionsPage.selectTab({
+				container: actionsPage.statusMessagesTabs,
+				label: 'Success',
+			});
+
+			await expect(form.successStatusMessageInput).toBeVisible();
+
+			await checkHelperTooltip({
+				formElement: form.successStatusMessageInput,
+				page,
+				text: 'The user will see this message if the action is successful.',
+			});
+
+			await actionsPage.selectTab({
+				container: actionsPage.statusMessagesTabs,
+				label: 'Error',
+			});
+
+			await expect(form.errorStatusMessageInput).toBeVisible();
+
+			await checkHelperTooltip({
+				formElement: form.errorStatusMessageInput,
+				page,
+				text: 'The user will see this message if the action fails.',
+			});
+		});
+
+		await test.step('Filter icon selection ', async () => {
+			await form.addIconButton.click();
+
+			const icon: string = 'chip';
+
+			await form.selectIconModal.getByPlaceholder('Search').fill(icon);
+
+			const iconsList = form.selectIconModal.getByRole('list');
+
+			const iconsListItem = iconsList.getByText(icon, {exact: true});
+
+			await expect(iconsListItem).toBeVisible();
+
+			await expect(iconsList.getByRole('listitem')).toHaveCount(1);
+
+			await form.selectIconModal
+				.getByRole('button', {name: 'Close'})
+				.click();
+
+			expect(form.iconInput).toHaveValue('');
+		});
+
+		await test.step('Validate entire form on save', async () => {
+			await form.typeSelect.selectOption('Link');
+
+			await form.saveButton.click();
+
+			await checkRequired({
+				formElements: [form.labelInput, form.urlInput],
+				page,
+			});
+
+			await form.typeSelect.selectOption('Headless');
+
+			await form.saveButton.click();
+
+			await checkRequired({
+				formElements: [form.labelInput, form.headlessActionKeyInput],
+				page,
+			});
+		});
+
+		await test.step('Validate required form elements individually', async () => {
+			await form.typeSelect.selectOption('Link');
+
+			await form.labelInput.fill(getRandomString());
+			await form.labelInput.clear();
+
+			await form.urlInput.fill(getRandomString());
+			await form.urlInput.clear();
+
+			await checkRequired({
+				formElements: [form.labelInput, form.urlInput],
+				page,
+			});
+
+			await form.typeSelect.selectOption('Headless');
+
+			await form.headlessActionKeyInput.fill(getRandomString());
+			await form.headlessActionKeyInput.clear();
+
+			await checkRequired({
+				formElements: [form.headlessActionKeyInput],
+				page,
+			});
+		});
+	}
+);
+
+test(
+	'Create and edit item action of type "Async"',
```

Comment:
There is quite a lot of duplication in create & edit tests for different action types. But, I still opted to do it this way as we need to cover three types of tests that existed in Poshi:
1. Tests checking available fields for each action type.
2. Tests checking correctly saved values, both after creation and after edit.
3. Tests checking creation with minimum required fields.